cli/azd/cmd/container.go

```diff
@@ -375,18 +378,109 @@ func registerCommonDependencies(container *ioc.NestedContainer) {
 		},
 	)
 
+	// Clients
+
+	// TODO: File this registration in a reasonable place
 	container.MustRegisterSingleton(func(
-		ctx context.Context,
+		env *environment.Environment,
+	) environment.SubscriptionId {
+		return environment.SubscriptionId(env.GetSubscriptionId())
+	})
```

Comment:
I don't think we're adding enough value by introducing the `SubscriptionId` type alias here.  We're only ever pulling from the azd env here, so you might as well just inject `env` into the client resolver functions below.